src/tests/core/blocks/test_validate_block.c

```diff
@@ -0,0 +1,291 @@
+#include <stdlib.h>
+#include "minunit.h"
+#include "validate_block.h"
+#include "create_block.h"
+#include "mempool.h"
+#include "utxo_pool.h"
+#include "sign_tx.h"
+#include "blockchain.h"
+#include "wallet_pool.h"
+
+int tests_run = 0;
+
+Transaction *_make_tx() {
+  Transaction *tx;
+  Input *in;
+  Output *out;
+  mbedtls_ecdsa_context *key_pair;
+
+  tx = malloc(sizeof(Transaction));
+  in = malloc(sizeof(Input));
+  out = malloc(sizeof(Output));
+
+  memset(in, 0, sizeof(Input));
+  key_pair = gen_keys();
+  in->pub_key = &(key_pair->private_Q);
+
+  memset(out, 0, sizeof(Output));
+
+  tx->num_inputs = 1;
+  tx->inputs = in;
+  tx->num_outputs = 1;
+  tx->outputs = out;
+
+  return tx;
+}
+
+void _fill_mempool(){
+  Transaction *input_tx, *tx1;
+  input_tx = _make_tx();
+  input_tx->outputs[0].amt = 100;
+  utxo_pool_init();
+  utxo_pool_add(input_tx, 0);
+
+  tx1 = _make_tx();
+  hash_tx(tx1->inputs[0].prev_tx_id, input_tx);
+  tx1->inputs[0].prev_utxo_output = 0;
+  tx1->outputs[0].amt = 90; 
+  mempool_init();
+  mempool_add(tx1);
+}
+
+static char  *test_coinbase_tx() {
+  _fill_mempool();
+  Block *test_block = create_block_alloc();
+  
+  mu_assert(
+    "Incorrect Coinbase TX",
```

Comment:
The error messages for the first set of tests are off. Here, it seems like we're assuming the block is valid, and are expecting the validation to return that the block is valid. Then, if the assert fails it means the validation function was wrong, not the block. But, the current error messages implicate the block creation.

(This one is pretty minor, ignore if needed)